jccli/jc_api_v1.py

```diff
@@ -180,3 +181,68 @@ def set_user(self, username, attributes):
             body=Systemuserput(**attributes)
         )
         return api_response.to_dict()
+
+    def search_systems(self, filter={}):
+        """
+        Search for users on JumpCloud.
+
+        :param filter: (dict) an object used to filter search results for various fields. E.g.: `{"firstname": "David"}`
+        :return:
+        """
+        query_filter = {'and': []}
+        for field, value in filter.items():
+            query_filter['and'].append({field: value})
+        if not filter:
+            query_filter = None
+
+        try:
+            api_response = self.search_api.search_systems_post(
+                content_type='application/json',
+                accept='application/json',
+                body={
+                    'filter': query_filter
+                }
+            )
+            systems = [system.to_dict() for system in api_response.results]
+            return systems
+        except ApiException as error:
+            raise "Exception when calling SystemusersApi->systems_search: %s\n" % error
+
+    def get_system(self, hostname):
+        """
+        Get detail view of a system.
+        :param hostname:
+        :return: system properties dict
+        """
+        systems = self.systems_api.systems_list(
```

Comment:
maybe instead of list and find you could use the search system api instead?  https://github.com/TheJumpCloud/jcapi-python/blob/master/jcapiv1/docs/SearchApi.md#search_systems_post

Reply:
changed